packages/shared/src/components/auth/OnboardingRegistrationForm.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {
+  ReactElement,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { useMutation } from 'react-query';
+import { checkKratosEmail } from '../../lib/kratos';
+import { AuthFormProps, providerMap } from './common';
+import OrDivider from './OrDivider';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import { AuthEventNames, AuthTriggersOrString } from '../../lib/auth';
+import { ExperimentWinner } from '../../lib/featureValues';
+import { Button } from '../buttons/Button';
+import AuthForm from './AuthForm';
+import { TextField } from '../fields/TextField';
+import MailIcon from '../icons/Mail';
+import { IconSize } from '../Icon';
+import Alert, { AlertParagraph, AlertType } from '../widgets/Alert';
+
+interface OnboardingRegistrationFormProps extends AuthFormProps {
+  onExistingEmail?: (email: string) => unknown;
+  onSignup?: (email: string) => unknown;
+  onProviderClick?: (provider: string, login?: boolean) => unknown;
+  targetId?: string;
+  isLoginFlow?: boolean;
+  logInTitle?: string;
+  signUpTitle?: string;
+  trigger: AuthTriggersOrString;
+  isReady: boolean;
+}
+
+const OnboardingRegistrationForm = ({
+  onSignup,
+  onExistingEmail,
+  onProviderClick,
+  targetId,
+  isReady,
+  trigger,
+}: OnboardingRegistrationFormProps): ReactElement => {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [shouldLogin, setShouldLogin] = useState(false);
+  // const { displayToast } = useToastNotification();
+  const [registerEmail, setRegisterEmail] = useState<string>(null);
+  const { mutateAsync: checkEmail } = useMutation((emailParam: string) =>
+    checkKratosEmail(emailParam),
+  );
+
+  useEffect(() => {
```

Comment:
I think that sounds like a good idea. I do feel like the entire auth could get a second look at see what is needed and not needed anymore too.

Reply:
Yes 100% it was very big because of all the different experiment implementations.
I think in the current scope we can downsize the logic quite a bit.